elements/nuxeo-document-bulk-actions/nuxeo-edit-documents-button.js

```diff
@@ -0,0 +1,613 @@
+      // move the `label` to the bulk widget
+      if (boundElement.label) {
+        bulkWidget.label = boundElement.label;
+        boundElement.label = null;
+      }
+      // move the `required` to the bulk widget
+      if (boundElement.required) {
+        bulkWidget._required = true;
+        boundElement.required = null;
+      }
+    });
+  }
+
+  /**
+   * Gets the bulk widget that is structurally above the provided `node`, recursively.
+   */
+  _getBulkWidget(node) {
+    if (!node.parentNode) {
+      return null;
+    }
+    const parent = node.parentNode;
+    return parent.tagName.toLowerCase() === 'nuxeo-bulk-widget' ? parent : this._getBulkWidget(parent);
+  }
+
+  /**
+   * Gets the element bound to a property that resides inside the provided `node` structure, recursively.
+   */
+  _getBoundElement(node, layoutTemplateInfo) {
+    // if there's no boundNodes information yet, create it by finding all the nodes that have bindings
+    if (!layoutTemplateInfo.boundNodes) {
+      layoutTemplateInfo.boundNodes = [];
+      for (let i = 0; i < layoutTemplateInfo.nodeList.length; i++) {
+        if (layoutTemplateInfo.nodeInfoList[i].bindings) {
+          layoutTemplateInfo.boundNodes.push(layoutTemplateInfo.nodeList[i]);
+        }
+      }
+    }
+    // if the widget itself is bound to a property, return it
+    if (layoutTemplateInfo.boundNodes.includes(node)) {
+      return node;
+    }
+    // keep searching recursively through the children (complex structure)
+    return Array.from(node.children).find((child) => this._getBoundElement(child, layoutTemplateInfo));
+  }
+
+  /**
+   * Returns whatever element is bound to the provided `boundPath`.
+   */
+  _getBoundElementFromPath(boundPath) {
+    return this.$$('nuxeo-layout')._getBoundElements(boundPath)[boundPath];
+  }
+
+  /**
+   * Clears the value of a property that is represented by the provided `boundPath`.
+   */
+  _clearValue(boundPath) {
+    if (!boundPath) {
+      return;
+    }
+    const bulkLayout = this.$$('nuxeo-layout').element;
+    const value = bulkLayout.get(boundPath);
+    if (Array.isArray(value)) {
+      if (value.length !== 0) {
+        bulkLayout.set(boundPath, []);
+      }
+    } else {
+      bulkLayout.set(boundPath, null);
+    }
+  }
+
+  /**
+   * Enables the provided `widget` for editing.
+   */
+  _enableWidget(widget) {
+    if (!widget) {
+      return;
+    }
+    if ('disabled' in widget) {
+      widget.disabled = false;
+    } else if ('readonly' in widget) {
+      widget.readonly = false;
+    } else if ('editable' in widget) {
+      widget.editable = true;
+    }
+  }
+
+  /**
+   * Disables the provided `widget` for editing.
+   */
+  _disableWidget(widget) {
+    if (!widget) {
+      return;
+    }
+    if ('disabled' in widget) {
+      widget.disabled = true;
+    } else if ('readonly' in widget) {
+      widget.readonly = true;
+    } else if ('editable' in widget) {
+      widget.editable = false;
+    }
+  }
```

Comment:
nit: We could have instead a single helper to set the "disabled" state to prevent duplicating a bunch of ifs, right? E.g.:
```suggestion
  _setWidgetDisable(widget, value) {
    if (!widget) {
      return;
    }
    if ('disabled' in widget) {
      widget.disabled = !!value;
    } else if ('readonly' in widget) {
      widget.readonly = !!value;
    } else if ('editable' in widget) {
      widget.editable = !value;
    }
  }
```

Reply:
Yes, we can.